upstream: https://sourceforge.net/tracker/index.php?func=detail&aid=1895615&group_id=33140&atid=407341
status: not necessary anymore at least from version 1.0.6 onwards where this patch reverse-applies cleanly

Index: acpid-1.0.3/ud_socket.c
===================================================================
--- acpid-1.0.3.orig/ud_socket.c	2003-11-17 14:24:58.000000000 -0700
+++ acpid-1.0.3/ud_socket.c	2005-06-27 14:44:17.785576106 -0700
@@ -58,7 +58,7 @@
 	while (1) {
 		int newsock = 0;
 		struct sockaddr_un cliaddr;
-		int len = sizeof(struct sockaddr_un);
+		socklen_t len = sizeof(struct sockaddr_un);
 
 		newsock = accept(listenfd, (struct sockaddr *)&cliaddr, &len);
 		if (newsock < 0) {
